### PR TITLE
Fix SDL does not increment count of rejections sync out of memory

### PR DESCRIPTION
**Pull request by LevchenkoS**
SDL should increment `count_of_rejections_sync_out_of_memory`
counter for an app, on sending result `OUT_OF_MEMORY`
##
@okozlovlux, @VVeremjova , @LitvinenkoIra, please, review.

### CODE REVIEW

File: src/components/application_manager/src/hmi_capabilities_impl.cc [hunk 1]
```diff
@@ -75,14 +75,18 @@ void InitCapabilities() {
...
       std::string("TEXT"), hmi_apis::Common_SpeechCapabilities::SC_TEXT));
-  tts_enum_capabilities.insert(std::make_pair(
-      std::string("SAPI_PHONEMES"), hmi_apis::Common_SpeechCapabilities::SAPI_PHONEMES));
-  tts_enum_capabilities.insert(std::make_pair(
-      std::string("LHPLUS_PHONEMES"), hmi_apis::Common_SpeechCapabilities::LHPLUS_PHONEMES));
-  tts_enum_capabilities.insert(std::make_pair(
-      std::string("SAPI_PHONEMES"), hmi_apis::Common_SpeechCapabilities::SAPI_PHONEMES));
-  tts_enum_capabilities.insert(std::make_pair(
-      std::string("PRE_RECORDED"), hmi_apis::Common_SpeechCapabilities::PRE_RECORDED));
+  tts_enum_capabilities.insert(
```
Comment (JackLivio) [inline src/components/application_manager/src/hmi_capabilities_impl.cc]: This PR should be to fix issue #1229 and should not include unrelated formatting changes

File: src/components/policy/policy_external/src/policy_table/types.cc [hunk 2]
```diff
@@ -1233,9 +1233,9 @@ Json::Value AppLevel::ToJsonValue() const {
       "minutes_in_hmi_limited", minutes_in_hmi_limited, &result__);
   impl::WriteJsonField(
       "minutes_in_hmi_background", minutes_in_hmi_background, &result__);
-  impl::WriteJsonField("minutes_in_hmi_none", minutes_in_hmi_none, &result__);
   impl::WriteJsonField(
-      "count_of_user_selections", count_of_user_selections, &result__);
```
Comment (JackLivio) [inline src/components/policy/policy_external/src/policy_table/types.cc]: Why was `count_of_user_selections` deleted here?

File: src/components/policy/policy_regular/src/policy_table/types.cc [hunk 3]
```diff
@@ -1003,7 +1003,41 @@ AppLevel::AppLevel(const Json::Value* value__)
...
+  impl::WriteJsonField("app_registration_language_vui",
+                       app_registration_language_vui,
+                       &result__);
+  impl::WriteJsonField(
+      "minutes_in_hmi_limited", minutes_in_hmi_limited, &result__);
+  impl::WriteJsonField(
+      "minutes_in_hmi_background", minutes_in_hmi_background, &result__);
+  impl::WriteJsonField(
+      "minutes_in_hmi_background", minutes_in_hmi_background, &result__);
+  impl::WriteJsonField("minutes_in_hmi_none", minutes_in_hmi_none, &result__);
```
Comment (JackLivio) [inline src/components/policy/policy_regular/src/policy_table/types.cc]: Was this just copied from the external policies types.cc? There is a duplicate `minutes_in_hmi_none`. Also just because these behavior exists in external policies, should it also exist in regular policies?

File: src/components/application_manager/src/hmi_capabilities_impl.cc (hunk 1 above)
Comment (LevchenkoS) [inline src/components/application_manager/src/hmi_capabilities_impl.cc]: @JackLivio, commit with coding style fix has been removed.

File: src/components/policy/policy_external/src/policy_table/types.cc (hunk 2 above)
Comment (LevchenkoS) [inline src/components/policy/policy_external/src/policy_table/types.cc]: @JackLivio, fixed in b9a7d78

File: src/components/policy/policy_regular/src/policy_table/types.cc (hunk 3 above)
Comment (LevchenkoS) [inline src/components/policy/policy_regular/src/policy_table/types.cc]: @JackLivio, method `ToJsonValue()` of `AppLevel` should write all data of `AppLevel` object to `JSON`. So, there, were added fields which been missed.

File: src/components/policy/policy_regular/src/policy_table/types.cc [hunk 4]
```diff
@@ -1003,7 +1003,41 @@ AppLevel::AppLevel(const Json::Value* value__)
...
+  impl::WriteJsonField("count_of_rpcs_sent_in_hmi_none",
+                       count_of_rpcs_sent_in_hmi_none,
+                       &result__);
+  impl::WriteJsonField("count_of_removals_for_bad_behavior",
+                       count_of_removals_for_bad_behavior,
+                       &result__);
+  impl::WriteJsonField("count_of_tls_errors", count_of_tls_errors, &result__);
+  impl::WriteJsonField("count_of_run_attempts_while_revoked",
+                       count_of_run_attempts_while_revoked,
+                       &result__);
```
Comment (jacobkeeler) [inline src/components/policy/policy_regular/src/policy_table/types.cc]: What issue is this solving? This change seems unrelated to this PR.

File: src/components/policy/policy_external/src/policy_table/types.cc [hunk 5]
```diff
@@ -1253,7 +1253,7 @@ Json::Value AppLevel::ToJsonValue() const {
   impl::WriteJsonField("count_of_removals_for_bad_behavior",
                        count_of_removals_for_bad_behavior,
                        &result__);
-  impl::WriteJsonField("count_of_TLS_errors", count_of_tls_errors, &result__);
+  impl::WriteJsonField("count_of_tls_errors", count_of_tls_errors, &result__);
```
Comment (jacobkeeler) [inline src/components/policy/policy_external/src/policy_table/types.cc]: Why this change? Again, seems out of scope.

File: src/components/policy/policy_regular/src/policy_table/types.cc (hunk 4 above)
Comment (LevchenkoS) [inline src/components/policy/policy_regular/src/policy_table/types.cc]: @jacobkeeler, it's add missing fields to `JSON`
Yes it's not directly related to this issue, but I think that it is still important one

File: src/components/policy/policy_external/src/policy_table/types.cc (hunk 5 above)
Comment (LevchenkoS) [inline src/components/policy/policy_external/src/policy_table/types.cc]: @jacobkeeler, verified the name with PT Data Dictionary. Proper naming is `count_of_TLS_errors`.
Fixed in 2d4aa21

File: src/components/policy/policy_external/src/policy_table/types.cc [hunk 6]
```diff
@@ -1216,7 +1216,7 @@ AppLevel::AppLevel(const Json::Value* value__)
           impl::ValueMember(value__, "count_of_rpcs_sent_in_hmi_none"))
     , count_of_removals_for_bad_behavior(
           impl::ValueMember(value__, "count_of_removals_for_bad_behavior"))
-    , count_of_tls_errors(impl::ValueMember(value__, "count_of_tls_errors"))
+    , count_of_tls_errors(impl::ValueMember(value__, "count_of_TLS_errors"))
```
Comment (jacobkeeler) [inline src/components/policy/policy_external/src/policy_table/types.cc]: Now I question how this ever worked if this value was different than the one in the `ToJsonValue` function.

**Comment by jacobkeeler** [comment]
I have been able to verify that this fix is working, so after the unrelated changes are moved, I will be able to approve this PR

**Comment by LevchenkoS** [comment]
@jacobkeeler, fix for PT Snapshot moved to separate PR #1537
